resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -123,14 +123,18 @@ public void operationComplete(Future<T> future) throws Exception {
                     }
                 }
             });
-            int dots = 0;
-            for (int idx = hostname.length() - 1; idx >= 0; idx--) {
-                if (hostname.charAt(idx) == '.' && ++dots >= parent.ndots()) {
-                    internalResolve(promise);
-                    return;
+            if (parent.ndots() == 0) {
+                internalResolve(promise);
+            } else {
+                int dots = 0;
+                for (int idx = hostname.length() - 1; idx >= 0; idx--) {
+                    if (hostname.charAt(idx) == '.' && ++dots >= parent.ndots()) {
+                        internalResolve(promise);
+                        return;
+                    }
                 }
+                promise.tryFailure(new UnknownHostException(hostname));
             }
-            promise.tryFailure(new UnknownHostException(hostname));
         }
```

Comment:
no, it's the same behavior than in the foreach loop that returns immediately after calling internalResolve.